bin/populate_wtsi_irods_groups.pl

```diff
@@ -117,68 +106,40 @@ sub _uid_to_irods_uid {
   return grep {/^\Q$u\E#/smx} @public;
 }
 
-Readonly::Scalar my $GROUP_SECONDARY_MEMBERS_FIELD_INDEX => 3;
-my%ug2id; #cache of group to users - populate here
-my%gid2group;
-my $num_group_lines = 0;
-open my$gfh, q(-|), q(getent group) or
-  $log->logcroak("Opening pipe to getent group failed: $ERRNO");
-while(<$gfh>){
-  $num_group_lines++;
-  chomp;
-  $log->debug("getent group: ", $_);
-  my@F=split /:/smx;
-  my$users=$ug2id{$F[0]}||=[];
-  push @{$users}, split /,/smx, $F[$GROUP_SECONDARY_MEMBERS_FIELD_INDEX]||q(); #fill with secondary groups for users
-  $gid2group{$F[2]}=$F[0];
-}
-close $gfh or
-  $log->logcroak("Closing pipe to getent group failed: $ERRNO");
-
-if ($group_min_record_count and $num_group_lines < $group_min_record_count) {
-  $log->logcroak("Output of 'getent group' appears truncated ",
-                 "($num_group_lines lines)");
-}
-
-Readonly::Scalar my $PASSWD_PRIMARY_GID_FIELD_INDEX => 3;
-
-my $num_passwd_lines = 0;
-open my$pfh, q(-|), q(getent passwd) or
-  $log->logcroak("Opening pipe to getent passwd failed: $ERRNO");
-while(<$pfh>){
-  $num_passwd_lines++;
-  chomp;
-  $log->debug("getent passwd: ", $_);
-  my@F=split /:/smx;
-  push @{$ug2id{$gid2group{$F[$PASSWD_PRIMARY_GID_FIELD_INDEX]}||=q()}},$F[0]; #fill with primary group for users - empty strong used if no group found for gid
-}
-close $pfh or
-  $log->logcroak("Closing pipe to getent passwd failed: $ERRNO");
-
-if ($passwd_min_record_count and $num_passwd_lines < $passwd_min_record_count) {
-  $log->logcroak("Output of 'getent passwd' appears truncated ",
-                 "($num_passwd_lines lines)");
+my $host = 'ldap.internal.sanger.ac.uk';
+my $ldap = Net::LDAP->new($host);
+
+$ldap->bind or $log->logcroak("LDAP failed to bind to '$host': ", $!);
+# Get group, gid and member uids from LDAP
+my ($group2uids, $gid2group) = find_group_ids($ldap);
+# Get uids and their primary gid from LDAP
+my $uid2gid = find_primary_gid($ldap);
+$ldap->unbind or $log->logwarn("LDAP failed to unbind '$host': ", $!);
+
+# For each uid, merge primary gid with secondary gids
+foreach my $uid (keys %{$uid2gid}) {
+  my $gid           = $uid2gid->{$uid};
+  my $primary_group = $gid2group->{$gid};
+  push @{$group2uids->{$primary_group}}, $uid;
 }
 
-foreach my $users (values%ug2id){
-  $users = [uniq @{$users}];
+foreach my $group (keys %{$group2uids}){
+  my @uids = uniq @{$group2uids->{$group}};
+  $group2uids->{$group} = \@uids;
+  $log->debug("Group '$group' membership ", join q(, ), @uids);
 }
 
-my $schema = npg_warehouse::Schema->connect;
-my $rs;
-if (@studies) {
-  $rs = $schema->resultset(q(CurrentStudy))->search({internal_id => \@studies});
-}
-else {
-  $rs = $schema->resultset(q(CurrentStudy));
-}
+my $mlwh = WTSI::DNAP::Warehouse::Schema->connect;
```

Comment:
DBIC dies on error, and the stack trace goes to STDERR. Trapping the error and re-throwing it with logging would help.